src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java

```diff
@@ -20,49 +22,66 @@
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final CategoryRepository categoryRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
 
     @Override
-    public BookDto save(CreateBookRequestDto requestDto) {
-        Book book = bookRepository.save(bookMapper.toModel(requestDto));
-        return bookMapper.toDto(book);
+    public BookResponseDto save(CreateBookRequestDto requestDto) {
+        List<Category> categories = categoryRepository
+                .findAllById(requestDto.getCategories());
+        if (categories.isEmpty()) {
+            throw new EntityNotFoundException("No categories with id "
+                    + requestDto.getCategories());
+        }
+        Book book = bookMapper.toModel(requestDto);
+        book.getCategories().addAll(categories);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.toDto(savedBook);
     }
 
     @Override
-    public BookDto findById(Long id) {
+    public BookResponseDto findById(Long id) {
         Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book by id: " + id)
         );
         return bookMapper.toDto(book);
     }
 
     @Override
-    public List<BookDto> getAll(Pageable pageable) {
+    public List<BookResponseDto> getAll(Pageable pageable) {
         return bookRepository.findAll(pageable).stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
     @Override
     @Transactional
-    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+    public BookResponseDto updateById(Long id, CreateBookRequestDto requestDto) {
         bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find and update book by id: " + id)
         );
+        List<Category> categories = categoryRepository
+                .findAllById(requestDto.getCategories());
+        if (categories.isEmpty()) {
+            throw new EntityNotFoundException("No categories with id "
+                    + requestDto.getCategories());
+        }
```

Comment:
You have the same logic in 2 places. You can consider extracting it to private method (but it's optional, decide what is more suitable for you)